pkg/controller/servicebinding/reconciler.go

```diff
@@ -113,6 +113,34 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return doneOnNotFound(err)
 	}
 
+	if sbr.GetDeletionTimestamp() != nil && sbr.GetOwnerReferences() != nil {
+		logger := logger.WithName("Deleting SBR when it has ownerReference")
+		logger.Debug("Removing resource finalizers...")
+		removeFinalizer(sbr)
+		if _, err := updateServiceBinding(r.dynClient, sbr); err != nil {
+			return requeueError(err)
+		}
+		return done()
+	}
+	if sbr.GetDeletionTimestamp() != nil {
+		logger := logger.WithName("unbind")
+		logger.Info("Executing unbinding steps...")
+		// here just build a servicebinder with necessary fields to do unbind
+		ensureDefaults(sbr.Spec.Application)
+		sb := &serviceBinder{
```

Comment:
could we use `buildServiceBinder()` here? How is this instance different from the existing instance, i.e. why `unbind` cannot be performed?

Reply:
@pedjak 
In current reconcile process, to do the unbind, it needs to `buildServiceContexts` and `buildBinding`, and `buildServiceBinder`. When the backing service has been deleted and it tries to delete the SBR, `buildServiceContexts` fail and requeue, then fail and requeue..... The SBR can not be deleted.

The reason here I did not use the `buildServiceBinder` is that it needs the result `buildServiceContexts` and `buildBinding`, which will fall as I mentioned above. And for unbind, I think there is no need to `buildServiceContexts` and `buildBinding` because the results of them are not necessary to do the unbind.

Reply:
@qibobo thanks for explanation. If I understand correctly, the issue here is that `buildServiceContexts()`is not aware of the fact that service binding is about to be deleted. However, it would be good that the change inside `Reconcile()` function is as small as possible. Hence, how about this: instead of creating `servicebinder` manually, how about that we use `buildServiceBinder` as before, but we creating empty service context if we recognize that deletion is in progress? Something like:

```golang
serviceCtxs, err := buildServiceContexts(...)
if err != nil {
   if sbr.GetDeletionTimestamp() != nil {
      serviceCtxs = serviceContextList{}
   } else {
	//handle service not found error
	if k8serrors.IsNotFound(err) {
        ...
        }
```

I believe this is the only change we need to perform in the reconcile function.